File: server/src/main/java/com/ps/server/dto/CreatePlayerDTO.java
package com.ps.server.dto;

import com.ps.server.Logic.Color;
import com.ps.server.enums.PlayerType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CreatePlayerDTO {
    private String username;
    private Color color;
    private PlayerType playerType;
}

File: client/src/app/DTO/createPlayerDTO.ts
import {Color} from "../_models/color";
import {PlayerType} from "../_models/playerType";


/** @ignore*/
export class CreatePlayerDTO {
    /** */
    public username: string;
    /** */
    public color: Color;
    /** */
    public playerType: PlayerType;

    /** @ignore*/
    constructor(username: string, color: Color, playerType: PlayerType) {
        this.username = username;
        this.color = color;
        this.playerType = playerType;
    }
}